ironfish/src/primitives/transaction.ts

```diff
@@ -58,11 +58,14 @@ export class Transaction {
    */
   withReference<R>(callback: (transaction: TransactionPosted) => R): R {
     const transaction = this.takeReference()
-    try {
-      return callback(transaction)
-    } finally {
+
+    const result = callback(transaction)
```

Comment:
Why not just await transaction?

Reply:
> Fixed this without making transaction.withReference async by creating a separate Promise.resolve() to wrap the result and calling finally on that one.

It's functionally possible to use async/await here instead, but it'll be a larger code change because all of the getters on `transaction` use this function, and they'll all have to return e.g. `Promise<number>` instead of `number`, then that'll propagate out to everywhere else in the codebase. What do you think?

Reply:
I can buy the reasoning.